src/typecheck.rs

```diff
@@ -22,6 +22,7 @@ pub enum TypeError {
 /// Helper function to create a type mismatch error
 fn type_mismatch(actual: &Type, expected: &Type, expr: &Expr) -> TypeError {
     debug!("In {expr:#?}");
+    error!("Type mismatch: expected {expected:?}, found {actual:?}");
```

Comment:
this shouldn't produce any error logging.

Reply:
I just don't want it at all. This error should be propagated to the main function. We discussed it before: **this is not an error in the typechecker so it shouldn't be logged as such.**, the type error is the return value of the typechecker.